yt/data_objects/static_output.py

```diff
@@ -1456,6 +1456,37 @@ def define_unit(self, symbol, value, tex_repr=None, offset=None, prefixable=Fals
         _define_unit(self.unit_registry, symbol, value, tex_repr=tex_repr,
                      offset=offset, prefixable=prefixable)
 
+    def rotate_frame_by(self, shift_angle, deg=False):
+        """
+        (experimental)
+        Shift azimuth coordinate 'theta' by *shift_angle*, counter-clockwise, and in radians by default.
+        This method is supported for polar and cylindrical geometries, not for spherical.
+
+        Parameters
+        ----------
+        shift_angle : float
+            angle by which coordinates should be shifted
+        deg : bool, optional
+            set to True to use a *shift_angle* in degrees
+        """
+        rotating_axes = {"polar": 1, "cylindrical": 2}
```

Comment:
So this is *almost* right, but we sometimes run into issues where we have different axes corresponding to different integer axis indices.

To get the right axis, you'll need to check `ds.coordinates.axis_id['theta']` to get the integer index of theta.

Reply:
I wasn't aware we had this, or that there would be edge cases. Thanks, this makes this a lot more robust !